Node/Generator/generators/app/index.js

```diff
@@ -24,7 +23,7 @@ module.exports = class extends Generator {
     const botName = this.props.botName;
     const extension = this.props.language === 'JavaScript' ? 'js' : 'ts';
     const launchSteps = extension === 'js' ? `node app.js` : `tsc\nnode app.js`;
-    const defaultDialog = this.props.dialog.split(' ')[0].toLowerCase();
+    const defaultDialog = this.props.dialog ? this.props.dialog.split(' ')[0].toLowerCase() : 'echo';
```

Comment:
'this.props.dialog.split' was returning an 'undefined' error when the dialog type was not specified by the user. this change allows the generator to default to 'echo' when no dialog type is specified and avoid and error (which were causing tests to fail). Tests have also been updated to pass in a value.